mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt

```diff
@@ -114,6 +123,37 @@ class GeneralSettings : BaseFragment() {
 
             recentFilesSwitch.mSwitch.isChecked = Preferences.isShowRecentFiles()
         }
+
+        val textJustificationSwitch = binding?.textJustificationSwitch
+
+        if (textJustificationSwitch != null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                textJustificationSwitch.mSwitch.setOnCheckedChangeListener { switch: CompoundButton?, isChecked: Boolean ->
+                    Preferences.setTextJustification(isChecked)
+                    refreshFragment()
+                }
+
+                textJustificationSwitch.mSwitch.isChecked = Preferences.isTextJustification()
+            } else {
+                textJustificationSwitch.hide()
+            }
+        }
+
+
+        val textSpacingSwitch = binding?.textSpacingSwitch
+        if (textSpacingSwitch != null) {
+            textSpacingSwitch.mSwitch.setOnCheckedChangeListener { switch: CompoundButton?, isChecked: Boolean ->
+                Preferences.setTextSpacing(isChecked)
+                refreshFragment()
```

Comment:
let's combine the code for textJustificationSwitch and textSpacingSwitch into a function to avoid redundancy.

Reply:
I meant using some method like this                                                                                                                       
    private fun setupSwitch(
    switch: CompoundButton?,
    isEnabled: () -> Boolean,
    setEnabled: (Boolean) -> Unit,
    refresh: () -> Unit
) {         
for both textJustificationSwitch and textSpacingSwitch